django_meili/management/commands/syncindex.py

```diff
@@ -11,19 +11,38 @@
 from django_meili._client import client as _client
 from django_meili.models import IndexMixin
 
+
+def batch_qs(qs, batch_size=1000):
+    """
+    Returns a (start, end, total, queryset) tuple for each batch in the given
+    queryset.
+    
+    Usage:
+        # Make sure to order your querset
+        article_qs = Article.objects.order_by('id')
+        for start, end, total, qs in batch_qs(article_qs):
+            print "Now processing %s - %s of %s" % (start + 1, end, total)
+            for article in qs:
+                print article.body
+    """
+    total = qs.count()
+    for start in range(0, total, batch_size):
+        end = min(start + batch_size, total)
+        yield (start, end, total, qs[start:end])
+
+
 class Command(BaseCommand):
     help = "Syncs the MeiliSearch index for the given model."
 
     def add_arguments(self, parser):
         parser.add_argument("model", type=str, help="The model to sync the index for. This should be in the format <app_name>.<model_name>")
+        parser.add_argument("batch_size", type=int, default=1000, help="The batch size you want to import in (default: 1000)")
```

Comment:
Let's have this be a flag with the default being the original behavior.
Instead the flag can determine the use of batching.

Reply:
I hear you, but then there are 2 code pathways? id argue for a single code pathway; ie. batching by default (as there is no impact on smaller querysets and no change for them) and by default larger qs are handled gracefully.

Agree re the option rather than arg

Reply:
Fair enough, although could we move the batching to some kind of configuration variable in the Django settings? I only say because I find that 1000 is both hard-coded and arbitrary. If we open it up that way it can mimic what Django already has in its `test` command with `--testrunner`.

Reply:
We can certainly do that but this is a command specific setting that only gets used for this specific command which argues against adding it to global settings? 

Arbitrary perhaps, but in my xp 1000 is a large enough number to get throughput without saturating memory with hydrated models?

Reply:
made batch_size optional ✅

Reply:
But in that same way Django's test command only uses `TEST_RUNNER` and its specific. If we move it to settings then it allows for env variables to be set or other interesting features to override it.

My main idea is that there's already things in Django that operate with it in a very specific way so there's precedence. It also gives a name to use in the code itself for the value.

Reply:
hmm test_runner is effectivly global as it impacts the global test runner. batch_size is only used in syncindex.

Sorry not sure I agree on this topic. But its your baby so

Reply:
Yeah, I appreciate your perspective and its definitely something I will consider if the settings begin to grow too large. Thank you for the dialogue!

Reply:
my pleasure, thanks for showing attention to detail and a focus on quality..

ps.. should apply black?

Reply:
Go for it. I haven't added the stamp to the README 😅 but I do use Black or Ruff for the code.

Reply:
added and applied black ✅